lib/ts/Scrubber.cc

```diff
@@ -0,0 +1,138 @@
+#include "Scrubber.h"
+#include "MemView.h"
+
+Scrubber::Scrubber(const char *config)
+{
+  bool state_expecting_regex = true;
+  char *regex                = nullptr;
+  char *replacement          = nullptr;
+  const ts::StringView delimiters("->;,", ts::StringView::literal);
+  ts::StringView text(config);
+
+  this->config = ats_strdup(config);
```

Comment:
Another option would be for Scrubber to take ownership of the memory and avoid the extra alloc/copy. Beyond that, since the entire string is already persistent, you can use `StringView` instances for the pieces and not allocate for them at all. E.g., `regex` can be a `StringView` that points in to `config`.